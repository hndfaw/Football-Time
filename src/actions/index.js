export const leaguesAction = leaguesData => ({
  type: 'SET_LEAGUES',
  leaguesData
})

export const setTodaysMatches = todayMatchesDate => ({
  type: 'SET_TODAYS_MATCHES',
  todayMatchesDate
})

export const setOneLeagueMatches = (leagueId, matches) => ({
  type: 'SET_ONE_LEAGUE_MATCHES',
  leagueId,
  matches
})