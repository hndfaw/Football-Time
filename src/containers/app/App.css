* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.img-background {
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  z-index: -10;
}

.color-background {
  width: 100vw;
  height: 100vh;
  background-image: linear-gradient(rgb(255, 255, 255), rgba(161, 148, 148, 0.514));
  position: fixed;
  top: 0;
  left: 0;
  z-index: -9;
}

.page-not-exist {
  margin-top: 25%;
  font-size: 18px;
  color: #000;
  padding-bottom: 20px;
}

.back-to-homepage {
  color: gray;
  font-size: 16px;
  border: 1px solid gray;
  text-decoration: none;
  border-radius: 3px;
  padding: 3px 7px;
}

.back-to-homepage:hover {
  background: rgb(245, 243, 243);
  color: rgb(119, 118, 118);
}

.loading {
  margin-top: 25%;
  height: 100px;
}