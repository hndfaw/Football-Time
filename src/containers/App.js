import React, {Component} from 'react';
import '../App.css';
import { connect } from 'react-redux';
import { fetchTodaysMatches } from '../apiCalls';
import { setTodaysMatches } from '../actions';
import  Header  from './header/Header'

class App extends Component {

  componentDidMount() {
    fetchTodaysMatches().then(data =>
      this.props.handleTodaysMatches(data)
      )
  }
  render() {
    return (
      <div className="App">
        <Header />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  handleTodaysMatches: data => dispatch(setTodaysMatches(data))
})

export default connect(null, mapDispatchToProps)(App);
