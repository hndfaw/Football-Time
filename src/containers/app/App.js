import React, {Component} from 'react';
import './App.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchTodaysMatches, fetchLeagues, fetchOneLeaguesMatches } from '../../apiCalls';
import { setTodaysMatches, leaguesAction, setPremierLeague} from '../../actions';
import  Header  from '../header/Header';
import  TodaysMatches  from '../todaysMatches/TodaysMatches';
import { Route, Switch, NavLink } from 'react-router-dom';
import Leagues from '../leagues/Leagues';
import Home from '../../components/home/Home';
import loading from '../../images/loading.gif';
import MatchDetail from '../../components/matchDetail/MatchDetail';


export class App extends Component {

   today = () => {
    let today = new Date();
    let dd = today.getDate();
    let mm = today.getMonth() + 1;
    let yyyy = today.getFullYear();
    if (dd < 10) {
      dd = '0' + dd;
    }
    if (mm < 10) {
      mm = '0' + mm;
    }
    return (yyyy + '-' + mm + '-' + dd);
  }

  componentDidMount() {
    fetchTodaysMatches(this.today()).then(data => {
      const cleanedData = this.cleanMatches(data.api.fixtures)
      this.props.handleTodaysMatches(cleanedData)
    })

    fetchLeagues().then(data =>
      this.props.handleLeagues(data.api.leagues))

      fetchOneLeaguesMatches(524).then(data => {
        const cleanedData = this.cleanMatches(data.api.fixtures)
        this.props.handlePremierLeague(cleanedData)
      })
  }


  cleanMatches = data => {
    const cleanedData = data.map(match => {

      const date = match.event_date.split("").slice(0, 10).join("")
        return {
          event_date: date,
          league_id: match.league_id,
          statusShort: match.statusShort,
          fixture_id: match.fixture_id,
          homeTeamName: match.homeTeam.team_name,
          homeTeamLogo: match.homeTeam.logo,
          awayTeamName: match.awayTeam.team_name,
          awayTeamLogo: match.awayTeam.logo,
          elapsed: match.elapsed,
          goalsHomeTeam: match.goalsHomeTeam,
          goalsAwayTeam: match.goalsAwayTeam,
          status: match.status
        }
      })
      return cleanedData;
  }

  render() {
    const { todaysMatches } = this.props
    const dataLoading = todaysMatches.length === 0
    return (
      <div className="App">
          <Header />
          <Switch>
            <Route exact path="/" render={() => (<Home />)} />
            <Route exact path="/todaysmatches" render={() => (
                dataLoading ?
                <img src={loading} className="loading" alt="loading icon" /> :
                <TodaysMatches cleanMatches={this.cleanMatches}/>
            )} />
            <Route exact path="/leagues/524" render={() => (
                <Leagues matches={this.props.league524} />
                )} />
            <Route exact path="/leagues/525" render={() => (
                <Leagues matches={this.props.league525}  />
                )} />
            <Route exact path="/leagues/530" render={() => (
                <Leagues matches={this.props.league530}  />
                )} />
            <Route exact path="/leagues/754" render={() => (
                <Leagues matches={this.props.league754}  />
                )} />
            <Route exact path="/leagues/775" render={() => (
                <Leagues matches={this.props.league775}  />
                )} />
            <Route exact path='/todaysmatches/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.todaysMatches.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route exact path='/leagues/524/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.league524.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route exact path='/leagues/525/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.league525.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route exact path='/leagues/530/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.league530.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route exact path='/leagues/754/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.league754.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route exact path='/leagues/775/:fixture_id' render={({ match}) => {
              const { fixture_id } = match.params;
              const foundMatch = this.props.league775.find(match => match.fixture_id === Number(fixture_id));
              return <MatchDetail {...foundMatch}/>
            }}/>
            <Route render={() => (
              <>
                <p className="page-not-exist">The page you’re looking for can’t be found.</p><NavLink to="/todaysmatches" className="back-to-homepage"> Back to Today's Matches</NavLink>
              </>
            )}/>
          </Switch>
      </div>
    );
  }
}

App.propTypes = {
  todaysMatches : PropTypes.array,
  league524 : PropTypes.array,
  league525 : PropTypes.array,
  league530 : PropTypes.array,
  league754 : PropTypes.array,
  league775 : PropTypes.array,
}

export const mapStateToProps = state => ({
  todaysMatches: state.todaysMatches,
  league524: state.league524,
  league525: state.league525,
  league530: state.league530,
  league754: state.league754,
  league775: state.league775,
})

export const mapDispatchToProps = dispatch => ({
  handleTodaysMatches: data => dispatch(setTodaysMatches(data)),
  handleLeagues: data => dispatch(leaguesAction(data)),
  handlePremierLeague: data => dispatch(setPremierLeague(data)),
})

export default connect(mapStateToProps, mapDispatchToProps)(App);