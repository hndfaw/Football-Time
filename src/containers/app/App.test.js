import React from 'react';
import { App , mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { setTodaysMatches, leaguesAction, setPremierLeague } from '../../actions';



describe('App', ()=> {
  it('should match the snapshot', ()=> {
    let wrapper = shallow(<App onDayDate={{date: '2019-01-01'}} todaysMatches={[{match: 'match'}]}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should dispatch with a setTodaysMatches action when handleTodaysMatches is called', () => {
    const mockDispatch = jest.fn();
    const mockMatches = [{}];
    const mockAction = setTodaysMatches(mockMatches);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleTodaysMatches(mockMatches);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

  it('should dispatch with a leaguesAction action when handleLeagues is called', () => {
    const mockDispatch = jest.fn();
    const mockMatches = [{}];
    const mockAction = leaguesAction(mockMatches);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleLeagues(mockMatches);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

  it('should dispatch with a setPremierLeague action when handlePremierLeague is called', () => {
  const mockDispatch = jest.fn();
  const mockMatch = [{match: 'match'}];
  const mockAction = setPremierLeague(mockMatch);
  const mappedProps = mapDispatchToProps(mockDispatch);
  mappedProps.handlePremierLeague(mockMatch);
  expect(mockDispatch).toHaveBeenCalledWith(mockAction);
})

})