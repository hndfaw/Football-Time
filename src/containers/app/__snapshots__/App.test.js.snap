// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/todaysmatches"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/leagues/524"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/leagues/525"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/leagues/530"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/leagues/754"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/leagues/775"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
